posthog/src/main/java/com/posthog/android/PostHogFeatureFlags.java

```diff
@@ -0,0 +1,246 @@
+    ) {
+        this.posthog = posthog;
+        this.flagCallReported = flagCallReported == null ? new HashMap() : flagCallReported;
+        this.reloadFeatureFlagsQueued = reloadFeatureFlagsQueued == null ? false : reloadFeatureFlagsQueued;
+        this.reloadFeatureFlagsInAction = reloadFeatureFlagsInAction == null ? false : reloadFeatureFlagsInAction;
+        this.featureFlagsLoaded = false;
+        this.logger = logger;
+        this.client = client;
+    }
+
+    public List<String> getFlags() {
+        return new ArrayList<String>(this.getFlagVariants().keySet());
+    }
+
+    public ValueMap getFlagVariants() {
+        Persistence persistence = this.posthog.persistenceCache.get();
+        return persistence.enabledFeatureFlags();
+    }
+
+    public Object getFeatureFlag(final @NonNull String key, final @Nullable Object defaultValue, final @Nullable Map<String, Object> options) {
+        if (!this.featureFlagsLoaded) {
+            this.logger.error(null, "getFeatureFlag for key %s failed. Feature flags didn't load in time.", key);
+            return defaultValue;
+        }
+        Object flagValue = this.getFlagVariants().get(key);
+        if (options != null && (Boolean) options.get("send_event") && this.flagCallReported.get(key) == null) {
+            this.flagCallReported.put(key, true);
+            this.posthog.capture(
+                    "$feature_flag_called",
+                    new Properties()
+                            .putValue("$feature_flag", key)
+                            .putValue("$feature_flag_response", flagValue));
+        }
+        if (flagValue != null && flagValue != "") {
+            return flagValue;
+        }
+        return defaultValue;
+    }
+
+    public Boolean isFeatureEnabled(final @NonNull String key, final @Nullable Boolean defaultValue, final @Nullable Map<String, Object> options) {
+        if (!this.featureFlagsLoaded) {
+            this.logger.error(null, "isFeatureEnabled for key %s failed. Feature flags didn't load in time.", key);
+            return defaultValue;
+        }
+        Object value = this.getFeatureFlag(key, defaultValue, options);
+        if (value != null) {
+            if (value instanceof Boolean) {
+                return (Boolean) value;
+            }
+            // Multivariate case
+            return true;
+        }
+        return defaultValue;
+    }
+
+    /**
+     * Reloads feature flags asynchronously.
+     * <p>
+     * Constraints:
+     * <p>
+     * 1. Avoid parallel requests
+     * 2. Delay a few milliseconds after each reloadFeatureFlags call to batch subsequent changes together
+     * 3. Called once when building PostHogFeatureFlags instance (see build method)
+     */
+    public void reloadFeatureFlags() {
+        if (!this.reloadFeatureFlagsQueued) {
+            this.reloadFeatureFlagsQueued = true;
+            this.startReloadTimer();
+        }
+    }
+
+    private void startReloadTimer() {
```

Comment:
I find this method slightly misleading tbh. I thought it was responsible for periodically refreshing feature flags but its actually debouncing using a timer. I wonder if we couldn't call it `reloadFeatureFlagsWithDebounce` or something to clearly differentiate it.

Reply:
Thanks for the quick review here! It's a name I borrowed from posthog-js but this is so much better. Will change—